src/tests/functional/drive_stylesheet_merging_tests.js

```diff
@@ -0,0 +1,20 @@
+import { test } from "@playwright/test"
+import { assert } from "chai"
+import { getComputedStyle, hasSelector, nextBody } from "../helpers/page"
+
+test.beforeEach(async ({ page }) => {
+  await page.goto("/src/tests/fixtures/stylesheets/left.html")
+})
+
+test("navigating removes unused style elements", async ({ page }) => {
+  await page.locator("#go-right").click()
+  await nextBody(page)
+
+  assert.ok(await hasSelector(page, 'link[rel=stylesheet][href="/src/tests/fixtures/stylesheets/common.css"]'))
+  assert.ok(await hasSelector(page, 'link[rel=stylesheet][href="/src/tests/fixtures/stylesheets/right.css"]'))
+  assert.notOk(await hasSelector(page, 'link[rel=stylesheet][href="/src/tests/fixtures/stylesheets/left.css"]'))
+
+  assert.equal(await getComputedStyle(page, "p", "font-weight"), "200")
+  assert.equal(await getComputedStyle(page, "p", "font-family"), "serif")
```

Comment:
This is only asserting the common styles, we could also assert the new style.

Reply:
The `serif` is asserting the new style. And the `200` is asserting that the old style has been removed, allowing the common style to come through.

It's a bit cryptic though isn't it. I'll look for a clearer way to do this.

Reply:
I mean the styles in `left.css` vs `right.css`, the difference is the background color, isn't it?

Reply:
Ah I see, sorry I misunderstood what you meant.

I was checking those by asserting whether or not the stylesheet links were present. We could verify the computed styles of those as well, though. Doesn't hurt to do both.

Reply:
OK I've changed it to assert the presence of links, and of embedded CSS rules. And also to check the computed styles that should apply once only the common & right styles are applied. (Which is a bit awkward because it needs to test that the old ones are actually gone, not just overridden, so it uses a couple different properties to do so).

I think it seems a little clearer now, as well as more thorough. What do you think?

Reply:
Nice 👌